### ISSUE TITLE
Update mosaic-contracts to work with solidity 0.5

### ISSUE THREAD

**Issue by schemar** [OpenSTFoundation/mosaic-contracts#454]
Don't set `compilers: solc: version: "0.5.0"` in `truffle.json`,as it requires the machine that runs the test to be online for every test to check for the solc binary. We should use "native" or "docker" instead.

**Comment by schemar** [comment]
It seems truffle `v5.0.0` is incompatible to `solc v0.5.0` at the moment. I cannot run `truffle compile` if I set `compilers: solc: version: "0.5.0"` in `truffle.json`.
It seems we need to wait for official support.

Also note that this should not be the way we do the upgrade in the future, as it requires the machine that runs the test to be *online for every test to check for the solc binary.* We should use "native" or "docker" instead.

**Comment by hobofan** [comment]
Should be possible once https://github.com/trufflesuite/truffle/pull/1433 is merged and released.

**Comment by schemar** [comment]
Duplicate of #479
